ssd1306/ssd1306.c

```diff
@@ -189,12 +203,12 @@ void ssd1306_UpdateScreen(void) {
     //  * 32px   ==  4 pages
     //  * 64px   ==  8 pages
     //  * 128px  ==  16 pages
-    for(uint8_t i = 0; i < SSD1306_HEIGHT/8; i++) {
-        ssd1306_WriteCommand(0xB0 + i); // Set the current RAM page address.
-        ssd1306_WriteCommand(0x00);
-        ssd1306_WriteCommand(0x10);
-        ssd1306_WriteData(&SSD1306_Buffer[SSD1306_WIDTH*i],SSD1306_WIDTH);
-    }
+#if defined(SSD1306_USE_SPI)
+    ssd1306_WriteCommand(0xB0); // Set the current RAM page address.
+    ssd1306_WriteCommand(0x00);
+    ssd1306_WriteCommand(0x10);
+#endif
```

Comment:
Could you please submit this refactoring as a separate PR? Also, I don't understand why it's inside ifdef(DMA)...

Reply:
These three `ssd1306_WriteCommand()` is used to tell SSD1306 which address do you want to begin to write into. The three are necessary for SPI, but ont necessary for I2C.
Because I2C has function `HAL_I2C_Mem_Write_DMA()`, you can pass the GRAM start address to it, but I did not find such a function for SPI.